### PR TITLE
Network configuration. Resolves #10.

### LINKED ISSUE #10: Allow advanced network config for creating network (hurl new)
Rather than just allowing number of channels, orgs, users etc.. allow an advanced option for creating a network - possibly based on a config file or config json passed on command line.

This could enable better control over how the network is built - controlling names and org<->channel relationships better.

For example
```

{
    "organizations": {
        "my-organization": {
            "channels": ["channel1", "channel2"],
            "peers": 3,
            "users": ["user1", "user2"]
        },
       "a-nother-org": {
            "channels": ["channel1"],
            "peers": 2,
            "users": ["user3", "user4"]
       }
    }
}
```

### PR DESCRIPTION
**Pull request by J3SR0**

### CODE REVIEW

**Comment by diestrin** [comment]
Thank you very much @J3SR0 for this PR. It looks really good to me actually. One thing to notice, I see you're deleting some of the changes of my last PR in this one, probably a conflict where you resolved on your favor. I'll go through the code and put some annotations where I see problems like this.

File: src/command.ts [hunk 1]
```diff
@@ -13,22 +13,22 @@ function collect(val, memo) {
 }
 
 const tasks = {
-    async createNetwork(organizations?: string, users?: string, channels?: string,
-        path?: string, inside?: boolean, skipDownload?: boolean) {
```
Comment (diestrin) [inline src/command.ts]: `skipDownload` should be preserved

File: src/command.ts [hunk 2]
```diff
@@ -41,20 +41,21 @@ const tasks = {
...
-    .option('--skip-download', 'Skip downloading the Fabric Binaries and Docker images')
     // .option('-p, --peers <peers>', 'Peers per organization')
     .action(async (cmd: any) => {
         if (cmd) {
             await tasks.createNetwork(
+                cmd.network,
                 !cmd.organizations || (cmd.organizations <= 2) ? 2 : cmd.organizations,
                 !cmd.users || (cmd.users <= 1) ? 1 : cmd.users,
                 !cmd.channels || (cmd.channels <= 1) ? 1 : cmd.channels,
-                cmd.path, !!cmd.inside, !!cmd.skipDownload
```
Comment (diestrin) [inline src/command.ts]: `skipDownload` should be preserved

File: src/generators/downloadFabricBinaries.ts [hunk 3]
```diff
@@ -124,7 +124,7 @@ echo "Checking IMAGES"
 
 dockerFabricPull() {
     local FABRIC_TAG=$1
-    for IMAGES in peer orderer tools; do
+    for IMAGES in peer orderer ccenv tools; do
```
Comment (diestrin) [inline src/generators/downloadFabricBinaries.ts]: ccenv is not needed

File: src/generators/downloadFabricBinaries.ts [hunk 4]
```diff
@@ -134,7 +134,7 @@ dockerFabricPull() {
   
   dockerThirdPartyImagesPull() {
     local THIRDPARTY_TAG=$1
-    for IMAGES in couchdb; do
+    for IMAGES in couchdb kafka zookeeper; do
```
Comment (diestrin) [inline src/generators/downloadFabricBinaries.ts]: kafka and zookeper are not needed

File: src/generators/invokechaincode.sh.ts [hunk 5]
```diff
@@ -63,13 +72,14 @@ export class InvokeChaincodeShGenerator {
...
-                        l(response.response.payload.toString('utf8'));
-                    }
+                    console.log(response);
+                    //l(`Response from ${response.peer.name}`);
+                    //if (response.isProposalResponse) {
+                    //    l(JSON.stringify(response));
+                    //} else {
+                    //    // Good response
+                    //    l(response.response.payload.toString('utf8'));
+                    //}
```
Comment (diestrin) [inline src/generators/invokechaincode.sh.ts]: Why this change?

**Comment by J3SR0** [comment]
@diestrin My bad, it was not intentional. I did the modifications. Please let me know if you see anything wrong.

File: src/generators/invokechaincode.sh.ts (hunk 5 above)
Comment (J3SR0) [reply src/generators/invokechaincode.sh.ts]: I just forgot to clean up. Note that the "peer" field is not always available in the response object.

**Comment by diestrin** [comment]
Hey @J3SR0, this is an awesome PR! I just finished testing it with multiple configurations as well as with old functionality, worked like a charm.

A couple of things to notice for the records:
- Paths like `$HOME/hyperledger-fabric-network/artifacts/crypto-config/peerOrganizations/org1.hurley.lab/users/User1@org1.hurley.lab/msp` will indeed change the organization name to the one in the configuration, but not the user. The fabric crypto-config does not allow to specify the user name so it will still be consecutive. This is important to have in mind.
- A follow up for this PR would be to store the json file in the Hurly home folder, so we can set some defaults based on the current configuration, instead of having to pass the org name in the install/instantiate

Again, thank you very much since this is a good improvement for Hurley for complex setups. May I ask one question, are you part of our Discord community? if so what's your username?

**Comment by diestrin** [comment]
Closes #10

**Comment by J3SR0** [comment]
Thx @diestrin. Yes, I am already on the Discord, the name is "Jesro".
